src/bucket.js

```diff
@@ -99,8 +99,22 @@ export default class Bucket {
    * @return {Promise<Object, Error>}
    */
   setData(data, options={}) {
+    if (typeof data !== "object") {
+      throw new Error("A bucket object is required.");
+    }
+
+    const bucket = {...data, id: this.name};
+
+    // For default bucket, we need to drop the id from the data object.
+    const bucketId = bucket.id;
+    if (bucket.id === "default") {
+      delete bucket.id;
```

Comment:
ah the good 'ol question with backward compat and version support. tbh, idk. was it just me, i'd just support master :D